evm/vm/forks/sharding/vm_state.py

```diff
@@ -240,7 +240,10 @@ def add_transaction(self, transaction, computation, block):
         block.transaction_fee_sum += transaction_fee
 
         # Get trie roots and changed key-values.
-        tx_root_hash, tx_kv_nodes = make_trie_root_and_nodes(block.transactions)
+        tx_root_hash, tx_kv_nodes = make_trie_root_and_nodes(
```

Comment:
It strikes me that `make_trie_root_and_nodes` should require the trie class in order to prevent these types of mistakes/errors.